pkg/util/stats_tracker_test.go

```diff
@@ -0,0 +1,170 @@
+package util
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func setupStatsTracker(timeFrame time.Duration, bucketSize time.Duration) (*int64, StatsTracker) {
+	now := time.Now().UnixNano()
+	s := NewStatsTrackerWithTimeProvider(timeFrame, bucketSize, func() int64 {
+		return now
+	})
+	return &now, s
+}
+
+// TestMovingAvg TODO
+func TestMovingAvg(t *testing.T) {
+
+	now, s := setupStatsTracker(3*time.Second, time.Second)
+
+	assert.Equal(t, int64(0), s.MovingAvg())
+
+	s.Add(10)
+	assert.Equal(t, int64(10), s.MovingAvg())
+
+	*now += int64(time.Second)
+	s.Add(10)
+
+	assert.Equal(t, int64(10), s.MovingAvg())
+
+	*now += int64(time.Second)
+	s.Add(10)
+
+	assert.Equal(t, int64(10), s.MovingAvg())
+
+	*now += int64(time.Second)
+	s.Add(20)
+
+	assert.Equal(t, int64(12), s.MovingAvg())
+
+	*now += int64(time.Second)
+	s.Add(40)
+
+	*now += int64(time.Second)
+	s.Add(60)
+
+	*now += int64(time.Second)
+	s.Add(80)
+
+	*now += int64(time.Second)
+	assert.Equal(t, int64(60), s.MovingAvg())
+
+	// clear out all data
+	*now += int64(100 * time.Second)
+	assert.Equal(t, int64(0), s.MovingAvg())
+
+}
+
+func TestMovingAvgBigWindow(t *testing.T) {
+
+	now, s := setupStatsTracker(24*time.Hour, time.Hour)
+	then := *now + 12*int64(time.Hour)
+
+	for *now < then {
+		s.Add(10)
+		*now += int64(time.Second)
+	}
+	assert.Equal(t, int64(10), s.MovingAvg())
+
+	then = *now + 12*int64(time.Hour)
+
+	for *now < then {
+		s.Add(30)
+		*now += int64(time.Second)
+	}
+	// actually 19.99 but check for 19 because of truncation
```

Comment:
Note: What do you think about using https://godoc.org/github.com/stretchr/testify/assert#InDelta?

Reply:
while that would be ideal, `s.MovingAvg()` returns an int64 so the truncation occurs internally. I'll clarify the comment instead.